pilot/pkg/config/kube/gateway/inferencepool_collection.go

```diff
@@ -516,16 +518,20 @@ func translateShadowServiceToService(existingLabels map[string]string, shadow sh
 
 	// Create a new service object based on the shadow service info
 	svc := &corev1.Service{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Service",
+		},
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      shadow.key.Name,
 			Namespace: shadow.key.Namespace,
-			Labels: maps.MergeCopy(map[string]string{
+			Labels: map[string]string{
```

Comment:
I am also uncertain I understand this e2e. 
Can you explain what are we doing exactly and SSA implication? 

Would be useful to understand what are we overriding and what are we merging?

Reply:
~~So the implication for this particular piece of code is that Istio becomes the owner of all of this service's labels? Seems fine to me, but want to see if there's any other implications there~~ Oh actually I see the tests show that labels are preserved, nice